deep_qa/data/data_generator.py

```diff
@@ -108,12 +109,13 @@ def create_generator(self, dataset: IndexedDataset):
         Main external API call: converts an ``IndexedDataset`` into a data generator suitable for
         use with Keras' ``fit_generator`` and related methods.
         """
+        unpadded_dataset = deepcopy(dataset)
```

Comment:
This line needs to be inside the `if self.sort_every_epoch` block.  To be sure of this, make a test that fails with the current code, to be sure that the change actually fixes the test.  I don't think your test fails with current code.